### PR TITLE
feat(func): add gzip, flate compressor/decompressor

### PR DESCRIPTION
**Pull request by carlclone**

### CODE REVIEW

File: internal/compressor/flate.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	return g.buffer.Bytes(), nil
+}
+
+func newFlateDecompressor() (*flateDecompressor, error) {
+	return &flateDecompressor{}, nil
+}
+
+type flateDecompressor struct {
+	reader io.ReadCloser
```
Comment (ngjaying) [inline internal/compressor/flate.go:45]: In the https://pkg.go.dev/compress/flate reset example, it is possible to init a reader with nil data, I suggest to follow the code style there. In zlib, NewReaer(nil) has an error, that's why I coded like this.

**Review by ngjaying — COMMENTED** [review]
Please check the built binary size to see if it expands too much after importing these new dependencies. Run `Make` and check the bin/kuiperd size.

File: internal/compressor/flate.go (hunk 1 above)
Comment (carlclone) [reply internal/compressor/flate.go:45]: got it , and why is this test failed ? 

```
mock sink receive  map[email:Dddd id:1 name:test]
--- FAIL: TestFormat_Apply (5.51s)
    sink_node_test.go:315: 1 	result mismatch:
        
        exp=[0a047465737410011a0444646464]
        
        got=[]
        
The test bucket size is 6.
```

Comment (carlclone) [reply internal/compressor/flate.go:45]: > In the https://pkg.go.dev/compress/flate reset example, it is possible to init a reader with nil data, I suggest to follow the code style there. In zlib, NewReaer(nil) has an error, that's why I coded like this.

code modified , and the size is kuiper 7.6M and kuiperd 28.6M  , same as before

Comment (ngjaying) [reply internal/compressor/flate.go:45]: > got it , and why is this test failed ?
> 
> ```
> mock sink receive  map[email:Dddd id:1 name:test]
> --- FAIL: TestFormat_Apply (5.51s)
>     sink_node_test.go:315: 1 	result mismatch:
>         
>         exp=[0a047465737410011a0444646464]
>         
>         got=[]
>         
> The test bucket size is 6.
> ```

Ah, it is a problem of the test case which is synced by time.sleep now.. Not your problem

**Review by ngjaying — APPROVED** [review]

**Comment by carlclone** [comment]
will do

**Comment by ngjaying** [comment]
Closes #1689

**Pull request merged**